src/main/java/com/blackducksoftware/integration/hub/alert/channel/manager/ChannelEventFactory.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2017 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.alert.channel.manager;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.blackducksoftware.integration.hub.alert.datasource.entity.distribution.DistributionChannelConfigEntity;
+import com.blackducksoftware.integration.hub.alert.datasource.entity.global.GlobalChannelConfigEntity;
+import com.blackducksoftware.integration.hub.alert.digest.model.ProjectData;
+import com.blackducksoftware.integration.hub.alert.event.AbstractChannelEvent;
+
+@Component
+public class ChannelEventFactory<E extends AbstractChannelEvent, D extends DistributionChannelConfigEntity, G extends GlobalChannelConfigEntity> {
+    private final List<DistributionChannelManager<G, D, E>> channelManagers;
+
+    @Autowired
+    public ChannelEventFactory(final List<DistributionChannelManager<G, D, E>> channelManagers) {
+        this.channelManagers = channelManagers;
+    }
+
+    public AbstractChannelEvent createEvent(final Long commonDistributionConfigId, final String distributionType, final ProjectData projectData) {
+        for (final DistributionChannelManager<G, D, E> manager : channelManagers) {
+            final AbstractChannelEvent event = manager.createChannelEvent(projectData, commonDistributionConfigId);
```

Comment:
Shouldn't the manager determine if this type isApplicable?

Reply:
Yeah, I can make it work.